### DISCUSSION TITLE
Npc skin

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#1382] [Help] [answered]
Hey, I want to set a skin to a npc. I have created my npc by creating a fakeplayer. But when I set the skin to PlayerSkin.fromUsername("BlokJochem"), I got a steve skin. What am I doing wrong?

**Comment by FluxCapacitor2** [comment, marked as answer]
I would recommend looking at this discussion: https://github.com/Minestom/Minestom/discussions/1239#discussioncomment-3162438
To change the skin of a player-like entity, you have to send an AddPlayerPacket with the skin. For regular players, Minestom handles this for you, but for fake players you have to do it manually.

**Reply by DevBlokJochem** [reply]
I was searching for the keyword "npc", but that is exactly what I was looking for. Thanks!

**Reply by DevBlokJochem** [reply]
I see that you are doing 
```skin.textures(),skin.signature()```
How can I implement this? I have a PlayerSkin as skin.

**Reply by FluxCapacitor2** [reply]
> I see that you are doing `skin.textures(),skin.signature()` How can I implement this? I have a PlayerSkin as skin.

The `textures()` and `signature()` return strings. In my example, `skin` is a `net.minestom.server.entity.PlayerSkin`, so I am just getting the values from the record.

**Reply by DevBlokJochem** [reply]
![image](https://user-images.githubusercontent.com/71212639/188331322-eede6aeb-1985-4d69-b448-a24f91899551.png)
```
private fun onJoinRegister(event: PlayerLoginEvent) {
        event.player.sendPacket(addPlayerPacket)
    }
```

Sometimes I can see the npc right, but sometimes the skin doesnt work. This sometimes happen when I join with my BlokJochem account alone too.

**Reply by DevBlokJochem** [reply]
![image](https://user-images.githubusercontent.com/71212639/188331443-17078e7f-705c-4221-8baf-7a31f61b8777.png)
and now the npc works on both accounts.
